Mathlib/Order/OrderType.lean

```diff
@@ -0,0 +1,450 @@
+/-
+Copyright (c) _
+-/
+import Mathlib.SetTheory.Cardinal.Basic
+import Mathlib.SetTheory.Ordinal.Basic
+import Mathlib.Order.Category.LinOrd
+import Mathlib.Algebra.Order.Ring.Unbundled.Rat
+import Mathlib.Data.Real.Basic
+import Mathlib.Data.Sum.Order
+import Mathlib.Data.Prod.Lex
+import Mathlib.Logic.Equiv.Prod
+/-!
+# OrderTypes
+
+OrderTypes are defined as equivalences of linear orders under order isomorphism. They are endowed
+with a preorder, where an OrderType is smaller than another one there is an order embedding
+into it.
+
+## Main definitions
+
+* `OrderType`: the type of OrderTypes (in a given universe)
+* `OrderType.type α`: given a type `α` with a linear order, this is the corresponding OrderType,
+* `OrderType.card o`: the cardinality of an OrderType `o`.
+* `OrderType.addMonoid`: the additive monoid of OrderTypes.
+* `OrderType.mul o₁ o₂`: the product of two OrderTypes `o₁` and `o₂`.
+
+A pre order with a bottom element is registered on OrderTypes, where `⊥` is
+`0`, the OrderType corresponding to the empty type.
```

Comment:
You can write "order type" as in plaintext, or `OrderType` to clarify you're talking about code.
```suggestion
`0`, the order type corresponding to the empty type.
```